### PR TITLE
Prepare release 0.9.0rc1.

### PR DESCRIPTION
**Pull request by notoraptor**
This PR prepares the release `0.9.0rc1`.

Missing: update of `Vision / State` section in `Theano/doc/introduction.txt`, and a general re-reading.

PS: For `.mailmap`, there are still few duplicated names (when running `git log --format="%aN <%aE>" | sort -u`), but it is difficult to choose the right email for these people (e.g., some of them use either `gmail` email or an email from France or other country [probably an university email]).

@nouiz @lamblin .

### CODE REVIEW

File: NEWS.txt [hunk 1]
```diff
@@ -3,12 +3,82 @@ Release Notes
...
+
+This release extends the 0.9.0beta1 and announces the upcoming final release 0.9.
+
+Highlights (since 0.9.0beta1):
+ - Better integration of Theano+libgpuarray packages into conda distribution
+ - Better handling of Windows end-lines into C codes
+ - Better compatibility with NumPy 1.12
+ - Faster scan optimizations
+ - Fixed broadcast checking in scan
+ - Bug fixes related to merge optimizer
```
Comment (nouiz) [inline NEWS.txt]: Crash fix related to the merge optimizer and shape inference.

File: NEWS.txt [hunk 2]
```diff
@@ -3,12 +3,82 @@ Release Notes
...
+ - many other bug fixes and improvements
+ - Updated documentation
+
+ - Old GPU back-end:
+   * In MRG, replaced method `multinomial_wo_replacement()` with new method `choice()`
```
Comment (nouiz) [inline NEWS.txt]: This is not related to the gpu. This is a user interface. Put this in the section: "Interface changes"

File: NEWS.txt [hunk 3]
```diff
@@ -3,12 +3,82 @@ Release Notes
...
+
+ - New GPU back-end:
+   * Value of a shared variable is now set inplace
+
+A total of 24 people contributed to this release, see the list at the bottom.
```
Comment (nouiz) [inline NEWS.txt]: ... to this release since 0.9beta1 and X since 0.8.

Just to make it more clear that the first number is just since the beta and show the big number since the last release.

File: NEWS.txt [hunk 4]
```diff
@@ -3,12 +3,82 @@ Release Notes
...
+ - Added Abstract Ops for batch normalization that use cuDNN when available and pure Theano CPU/GPU alternatives otherwise
+ - Added new Theano flag cuda.enabled
+ - Added new Theano flag print_global_stats to print some global statistics (time spent) at the end
+
+Others:
+ - Split op now has C code for CPU and GPU
+ - "theano-cache list" now includes compilation times
+
+Other more detailed changes:
+ - Changed optdb.max_use_ratio to 6
```
Comment (nouiz) [inline NEWS.txt]: Remove this. This is too much a detail and the only consequence is less warning output during tests.

**Comment by notoraptor** [comment]
For `Vision State`, I plan to update just the date.

Maybe we could also add something about OpFromGraph, like:
`* We support the creation of new operations from graphs at runtime, which allow gradients overload for every input` ?

**Comment by nouiz** [comment]
Good idea. append to that phrase something like "and allow inlining at the start of compilation"

**Comment by notoraptor** [comment]
Since there are some updates on the master branch, I think I should rebase this PR, isn't it, @nouiz ? Or I may wait to see if master branch will be updated again in next hours/days ?

**Comment by notoraptor** [comment]
Updated, @nouiz .
